js/initialize.jsx

```diff
@@ -0,0 +1,147 @@
+'use strict'
+
+var React = require('react/addons')
+var ipc = window.require('remote').require('ipc')
+
+var $ = require('jquery-bf')
+var bs = require('bootstrap')
+
+var Initform = React.createClass({
+  displayName: 'Initform',
+  mixins: [React.addons.LinkedStateMixin],
+
+  getInitialState: function () {
+    return {
+      error: null,
+      directory: '',
+      initState: 'uninitialized',
+      keysize: 2048
+    }
+  },
+
+  componentDidMount: function () {
+    var t = this
+    ipc.on('default-directory', function (dir) {
+      t.setState({directory: dir})
+    })
+
+    ipc.on('initializing', function () {
+      t.setState({initState: 'initializing'})
+    })
+
+    ipc.on('initialization-error', function (err) {
+      t.setState({
+        error: err,
+        initState: 'uninitialized'
+      })
+    })
+
+    ipc.on('initialization-complete', function () {
+      t.setState({initState: 'initialized'})
+    })
+  },
+
+  setKeySize: function (size) {
+    var t = this;
+    return function () {
+      t.setState({keysize: size})
+    }
+  },
+
+  initialize: function () {
+    ipc.emit('initialize', {
+      directory: this.state.directory,
+      keysize: this.state.keysize
+    })
+  },
+
+  render: function () {
+    var t = this
+    var initializing = (t.state.initState === 'initializing')
+    var initialized = (t.state.initState === 'initialized')
+    var uninitialized = (t.state.initState === 'uninitialized')
+
+    var description = 'Before starting, ipfs needs to generate a keypair, and create a directory to live in.'
```

Comment:
all of this should probably live on html pages/templates that we can easily tweak. we'll want to play with all this stuff quite a bit as time goes on.